tests/integration/test_nidcpower_reservation.py

```diff
@@ -0,0 +1,117 @@
+import pathlib
+from typing import NamedTuple, TypeVar, Union
+
+import pytest
+
+from ni_measurementlink_service.session_management import (
+    Connection,
+    PinMapContext,
+    SessionManagementClient,
+    TypedConnection,
+)
+from tests.utilities.pin_map_client import PinMapClient
+
+
+def test___single_session_reserved___initialize_nidcpower_session___single_session_created(
+    pin_map_client: PinMapClient,
+    pin_map_directory: pathlib.Path,
+    session_management_client: SessionManagementClient,
+) -> None:
+    pin_map_name = "1Smu2ChannelGroup2Pin1Site.pinmap"
+    pin_names = ["Pin1"]
+    pin_map_id = pin_map_client.update_pin_map(pin_map_directory / pin_map_name)
+    pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0])
+    with session_management_client.reserve_session(
+        pin_map_context, pin_names, timeout=0
+    ) as reservation:
+        with reservation.initialize_nidcpower_session() as session_info:
+            assert session_info.session is not None
+
+        assert session_info.session_name == "DCPower1/0"
```

Comment:
Split the test body into 3 sections for arrange, act, and assert, separated by a single blank line. Do not comment the sections. The middle section (act) should call initialize_nidcpower_session. See https://dev.azure.com/ni/DevCentral/_wiki/wikis/AppCentral.wiki/20934/Regression-Test-Conventions for more details

To prevent Black from reformatting the blank lines, use a single ExitStack for the whole test:

```python
with ExitStack() as stack:
    ...
    reservation = stack.enter_context(session_management_client.reserve_session(...))

    session_info = stack.enter_context(reservation.initialize_nidcpower_session())

    assert ...
```

Reply:
I'm aware of having a blank line between those 3 sections in the test body, but as you mentioned Black removed them :( 
Thank you for suggesting, used the `ExitStack` to resolve it.